packages/esbuild-plugin/src/index.ts

```diff
@@ -239,18 +237,14 @@ function esbuildBundleSizeOptimizationsPlugin(
   return {
     name: "sentry-esbuild-bundle-size-optimizations-plugin",
     esbuild: {
-      setup({ onLoad }) {
-        onLoad({ filter: /./ }, async (args) => {
-          const source = await fs.promises.readFile(args.path, "utf8");
-
-          const contents = replaceBooleanFlagsInCode(source, replacementValues);
-
-          if (!contents) {
-            return null;
-          }
-
-          return { contents: contents.code, loader: "default" };
-        });
+      setup({ initialOptions }) {
+        const replacementStringValues: Record<string, string> = Object.entries(
+          replacementValues
+        ).reduce((acc, [key, value]) => {
+          return { ...acc, [key]: JSON.stringify(value) };
+        }, {});
```

Comment:
this is an accidental O(n^2) because of the spread in a loop, I don't think it matters a lot but since this is an easy fix we should just fix it.